### DISCUSSION TITLE
Unable to install Kimai

### DISCUSSION THREAD

**Discussion by Rockam** [kimai/kimai#2308] [Questions & Answers]
Hi, I'm trying to install Kimai on my Ubuntu 20.04 LTS via SSH.

I have installed all the server requirements and everything and I'm following the installation guide. However, when I reach the part where i have to run:

`bin/console kimai:install -n`

This error arises:
![kimai-error](https://user-images.githubusercontent.com/9072852/106602934-f3b4ac00-655d-11eb-97ce-3d853cd9fc0e.png)

Please, I've tried solutions all over the internet and I can't manage to make it work.

Hope you can help me, thanks in advance.

-----------------------------------

I'm using:

mariadb 15.1
php 7.4.3
apache2

**Comment by kevinpapst** [comment]
Did you edit .env and change the DATABASE_URL?

**Reply by Rockam** [reply]
Thank you so much for the quick response Kevin.

Yes I did, but checking that file again I had some issues with it. I managed to solve it and now it works! :) I'll continue with the next steps, I really liked the demo.

Thanks for your awesome work!!

**Reply by adrienromain** [reply]
Hello, 

I have exactly the same problem as @Rockam with the same error message and the same configuration : 
mysql ver 15.1
php 7.4.14

I modified .env like this :
DATABASE_URL=mysql://utilisateur:pass@127.0.0.1:3306/bdname


Can you help me please :)

**Reply by kevinpapst** [reply]
You are missing the PHP driver for mysql. 
You can first try to use `mysqli://utilisateur:pass@127.0.0.1:3306/bdname` and if that doesn't work as well, then you need to install a package like `php-mysql`. See here for an example: https://www.kimai.org/documentation/fresh-ubuntu-20.html#install-php-webserver-and-database

**Reply by adrienromain** [reply]
Thank you for the answer :)
I no longer have the error putting mysqli instead of mysql, but I get the following error :

_In MysqliConnection.php line 64:
 "Attempted to call function "mysqli_init" from the global namespace."_

Where can this error come from?

For information, I am on a Debian 10 server.

thank you in advance

**Reply by kevinpapst** [reply]
You change mysql to mysqli and seriously wonder where the error comes from?
Did you install what I said?

**Reply by adrienromain** [reply]
Thank you, indeed some packages were missing. I thought they had already installed with the plesk system I am using on my server.

We advance little by little ... The homepage is displayed but with the error "Something is wrong"
I checked the permissions which look good. I also delete the directories in /var/cache as explained in #1485 

I have this error in prod.log: 

php.CRITICAL: Uncaught Error: Argument 1 passed to Symfony\Component\HttpFoundation\Session\Storage\Handler\PdoSessionHandler::connect() must be of the type string, object given, called in /var/www/vhosts/artmo.cloud/time.artmo.cloud/kimai2/vendor/symfony/http-foundation/Session/Storage/Handler/PdoSessionHandler.php on line 270 {"exception":"[object] (TypeError(code: 0): Argument 1 passed to Symfony\\Component\\HttpFoundation\\Session\\Storage\\Handler\\PdoSessionHandler::connect() must be of the type string, object given, called in /var/www/vhosts/artmo.cloud/time.artmo.cloud/kimai2/vendor/symfony/http-foundation/Session/Storage/Handler/PdoSessionHandler.php on line 270 at /var/www/vhosts/artmo.cloud/time.artmo.cloud/kimai2/vendor/symfony/http-foundation/Session/Storage/Handler/PdoSessionHandler.php:436)"} []

Thank you in advance

**Reply by kevinpapst** [reply]
I have never seen that before. Maybe Plesk added some magic session optimization (eg store them in redis) through the general php configuration. Or you are using a wrong php version (difference between console and web).

**Reply by adrienromain** [reply]
thank you I will continue to search and I will come and give the info here if successful.

**Reply by adrienromain** [reply]
Hello,
I come back with some good news, I managed to get Kimai to work and resolve the error mentioned above.

In fact, the problem was with the version of the Symfony packages.

After running the command : 
_composer install --no-dev --optimize-autoloader_

I made an update with this :
_composer update_

Many packs (including Symfony) have been updated.

And then everything is back to normal.


I just still had a problem with DATABASE_URL in .env and with the special characters of the password.
I encode the password with urlencode from here https://fr.functions-online.com/urlencode.html

Then i wrote down the encoded password i got in .env but i still had a malformed url problem.

After several searches I modified /config/packages/doctrine.yaml line 14:

I replaced 
'% env (resolve: DATABASE_URL)%' 
with 
'% env (DATABASE_URL)%' 
and everything worked.

Hope this will benefit others;)

Enjoy with time tracking with this great app !! Thank you !

**Reply by kevinpapst** [reply]
Glad that it works now 👍  That one is interesting:

> env (resolve: DATABASE_URL)'

I will check that.

In case someone else reads this.... running `composer update` leaves you in an unknown and untested state. It is NOT recommended that you try to fix problems by updating required libraries. You are on your own when running into problems AND you are on your own when it comes to the next git pull & kimai update.

**Reply by adrienromain** [reply]
thanks again for the support!
